### PR TITLE
Feature/employee expenses/ticket#59

### PR DESCRIPTION
**Pull request by JoshM0101**
# Pull Request Review Comments

## Overview
- **Purpose of the PR**:
    - [ ] Fix Employee Expenses Registration
    - [ ] Fix Employee Expenses List

---

_**Additional Comments for the Reviewer (Optional)**_

- Please pull from develop then migrate

### CODE REVIEW

File: backend/api/views.py [hunk 1]
```diff
@@ -856,45 +899,104 @@ class CreateEmployeeExpenses(generics.CreateAPIView):
...
-                    responses.append({"message": "Personnel Planning Data created successfully."})
-                else:
-                    return JsonResponse(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
-            except Projects.DoesNotExist:
-                return JsonResponse({"message": "Planning project not found."}, status=status.HTTP_404_NOT_FOUND)
-            except Employees.DoesNotExist:
-                return JsonResponse({"message": "User not found."}, status=status.HTTP_404_NOT_FOUND)
-            except Exception as e:
-                return JsonResponse({"messagessss": str(e)}, status=status.HTTP_400_BAD_REQUEST)
+    def create(self, request, *args, **kwargs):
```
Comment (edward-stalley-work) [inline backend/api/views.py:902]: Please add ALL validation messages:
Need modal/alert box like the other screens for the user.

It is possible to create an empty record. Should be validation to ensure all fields have been entered. Including year and month.

Screenshot 2024-10-15 at 10 37 43


Screenshot 2024-10-15 at 10 36 24

File: client/src/assets/scss/EmployeeExpenses/EmployeeExpensesList.scss [hunk 2]
```diff
@@ -81,53 +112,30 @@
                                             }
                                             .employeeExpensesList_txt0{
                                                 margin-bottom: 5px;
-                                                text-align: center;
```
Comment (edward-stalley-work) [inline client/src/assets/scss/EmployeeExpenses/EmployeeExpensesList.scss:84]: Is it possible to have alternate grey colors like the other list page rows.
Just to make each project more clear?

Like this color below ?
Screenshot 2024-10-15 at 10 46 51

Or  if that looks bad - something to help separate each project visually.

**Review by edward-stalley-work — CHANGES_REQUESTED** [review]
Thank you Joshua - please see the comments.

File: client/src/pages/EmployeeExpenses/EmployeeExpensesRegistration.tsx [hunk 3]
```diff
@@ -53,298 +58,339 @@ const EmployeeExpensesRegistration = () => {
...
+                        type='button'
+                        onClick={removeEmployeeContainer}
+                      />
+                    </div>
+                    <div className='employeeExpensesRegistration_btn-subcancel'>
+                      <button type='button' className='button is-light'>
+                        {translate('cancel', language)}
+                      </button>
+                      <button type='submit' className='button is-info'>
+                        {translate('submit', language)}
```
[the remark this answers is not in the document]
Comment (edward-stalley-work) [reply client/src/pages/EmployeeExpenses/EmployeeExpensesRegistration.tsx:382]: Fixed. (please ignore this Joshua)
Issue was due to not dropping DB before running "makemigrations api" and "migrate".
Works fine now.

File: client/src/components/Localhost/Localhost.tsx [hunk 4]
```diff
@@ -13,7 +13,7 @@ export const fetchApi = async (
       throw new Error(`Error: ${response.statusText}`)
     }
     const data = await response.json()
-    onSuccess(data)
+    onSuccess(data)
```
[the remark this answers is not in the document]
Comment (JoshM0101) [reply client/src/components/Localhost/Localhost.tsx:16]: fixed

File: client/src/pages/EmployeeExpenses/EmployeeExpensesRegistration.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (JoshM0101) [reply client/src/pages/EmployeeExpenses/EmployeeExpensesRegistration.tsx:382]: ignored

File: client/src/pages/EmployeeExpenses/EmployeeExpensesList.tsx [hunk 5]
```diff
@@ -127,10 +242,181 @@ const EmployeeExpensesList: React.FC = () => {
                   <div className='employeeExpensesList_table_cont'>
                     <div className='columns is-mobile'>
                       <div className='column'>
+                      {isEditing ? (
```
[the remark this answers is not in the document]
Comment (JoshM0101) [reply client/src/pages/EmployeeExpenses/EmployeeExpensesList.tsx:245]: fixed

File: client/src/pages/EmployeeExpenses/EmployeeExpensesRegistration.tsx [hunk 6]
```diff
@@ -53,298 +58,339 @@ const EmployeeExpensesRegistration = () => {
...
-                 </div>
-         </div>
-     </div>
- </div>
-  );
+                            </div>
+                          ))}
+                          <div className='employeeExpensesRegistration_button-box'>
+                            <Btn
+                              label={translate('add', language)}
```
[the remark this answers is not in the document]
Comment (JoshM0101) [reply client/src/pages/EmployeeExpenses/EmployeeExpensesRegistration.tsx:352]: fixed

File: backend/api/views.py (hunk 1 above)
Comment (JoshM0101) [reply backend/api/views.py:902]: fixed

File: client/src/assets/scss/EmployeeExpenses/EmployeeExpensesList.scss (hunk 2 above)
Comment (JoshM0101) [reply client/src/assets/scss/EmployeeExpenses/EmployeeExpensesList.scss:84]: fixed

File: client/src/pages/EmployeeExpenses/EmployeeExpensesList.tsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (edward-stalley-work) [reply client/src/pages/EmployeeExpenses/EmployeeExpensesList.tsx:248]: apologies i think this was an issue with my screen.
Possibly not necessary to have scroll on x but we can leave it for time being.

File: backend/api/views.py (hunk 1 above)
Comment (edward-stalley-work) [reply backend/api/views.py:902]: As it is easier to modify and track in the future can we use the JSON for general words for all validation messages. (english and japanese) 

naming them like this is fine

translate('employeeExpensesValidationText1', language)}

**Review by edward-stalley-work — APPROVED** [review]

File: backend/api/views.py (hunk 1 above)
Comment (edward-stalley-work) [reply backend/api/views.py:902]: This will be handled in ticket 94 (modals and validation updates)

**Pull request merged**